src/Tinker.jl

```diff
@@ -49,12 +66,28 @@ function init_gui(image::AbstractArray; name="Tinker")
         end
     end
 
-    # rectangle draw function
-    function drawrect(ctx, rect, color)
-        set_source(ctx, color)
-        rectangle(ctx, rect[1], rect[2], rect[3], rect[4])
-        stroke(ctx)
-    end;
+    # holds x zoom level
+    xzoom = map(zr) do r
```

Comment:
You might be able to define these simply as functions that work on a `ZoomRegion` object, rather than making them signals. In other words, something like
```julia
xzoom(r::ZoomRegion) = 100*r.fullview.x.right/(r.currentview.x.right-(r.currentview.x.left-1))
```